webapp/src/components/conference/conference.tsx

```diff
@@ -0,0 +1,323 @@
+            jwt: this.props.jwt,
+            noSSL,
+            parentNode: document.querySelector('#jitsiMeet'),
+            onload: () => {
+                this.setState({loading: false});
+                this.resizeIframe();
+            }
+        };
+        this.api = new (window as any).JitsiMeetExternalAPI(domain, options);
+        this.api.on('videoConferenceJoined', () => {
+            if (this.state.minimized) {
+                this.minimize();
+            }
+        });
+        this.api.on('readyToClose', () => {
+            this.close();
+        });
+        this.api.on('tileViewChanged', (event: {enabled: boolean}) => {
+            if (!this.state.minimized) {
+                this.setState({wasTileView: event.enabled});
+            }
+            this.setState({isTileView: event.enabled});
+        });
+        this.api.on('filmstripDisplayChanged', (event: {visible: boolean}) => {
+            if (!this.state.minimized) {
+                this.setState({wasFilmStrip: event.visible});
+            }
+            this.setState({isFilmStrip: event.visible});
+        });
+        this.api.executeCommand('subject', post.props.meeting_topic);
+    }
+
+    resizeIframe = () => {
+        if (this.api && this.props.post) {
+            const vw = this.getViewportWidth();
+            const vh = this.getViewportHeight();
+            const iframe = this.api.getIFrame();
+            iframe.style.width = this.state.minimized ? MINIMIZED_WIDTH + 'px' : vw + 'px';
+            iframe.style.height = this.state.minimized ? MINIMIZED_HEIGHT + 'px' : vh + 'px';
+        }
+    }
+
+    componentDidMount() {
+        window.addEventListener('resize', this.resizeIframe);
+        if (this.props.post) {
+            this.initJitsi(this.props.post);
+        }
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('resize', this.resizeIframe);
+        if (this.api) {
+            this.api.dispose();
+        }
+    }
+
+    componentDidUpdate(prevProps: Props, prevState: State) {
+        if (this.props.post) {
+            if (prevProps.post !== this.props.post) {
+                this.initJitsi(this.props.post);
+            }
+            if (prevState.minimized !== this.state.minimized) {
+                this.resizeIframe();
+            }
+        }
+    }
+
+    close = () => {
+        this.api.executeCommand('hangup');
+        setTimeout(() => {
+            this.props.actions.openJitsiMeeting(null, null);
+            this.setState({
+                minimized: false,
+                loading: true,
+                position: POSITION_BOTTOM,
+                wasTileView: true,
+                isTileView: true,
+                wasFilmStrip: true,
+                isFilmStrip: true
+            });
+        }, 200);
+    }
+
+    minimize = () => {
+        this.setState({minimized: true});
+        if (this.state.isTileView) {
+            this.api.executeCommand('toggleTileView');
```

Comment:
Should these toggles update the state, or the state is tracking a different thing?

Reply:
That toggles update the state in other place, using the listeners initialized by the `initJitsi` function.